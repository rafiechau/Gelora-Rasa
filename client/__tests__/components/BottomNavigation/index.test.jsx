// BottomBar.test.js
import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom';
import BottomBar from '@components/BottomNavigation';

// Mock Material UI Icons
jest.mock('@mui/icons-material/Person', () => () => <div>PersonIcon</div>);
jest.mock('@mui/icons-material/History', () => () => <div>HistoryIcon</div>);
jest.mock('@mui/icons-material/Event', () => () => <div>EventIcon</div>);
jest.mock('@mui/icons-material/LocationOn', () => () => <div>LocationOnIcon</div>);
jest.mock('@mui/icons-material/Group', () => () => <div>GroupIcon</div>);
jest.mock('@mui/icons-material/Category', () => () => <div>CategoryIcon</div>);

// Create a mock Redux store
const mockStore = {
  getState: () => ({}),
  subscribe: () => {},
  dispatch: jest.fn(),
};

describe('BottomBar Component', () => {
  const mockUser = { role: 2 };

  it('renders correctly', () => {
    const { getByTestId } = render(
      <Provider store={mockStore}>
        <MemoryRouter>
          <BottomBar user={mockUser} />
        </MemoryRouter>
      </Provider>
    );
    expect(getByTestId('bottom-navigation')).toBeInTheDocument();
  });

  it('matches the snapshot', () => {
    const { asFragment } = render(
      <Provider store={mockStore}>
        <MemoryRouter>
          <BottomBar user={mockUser} />
        </MemoryRouter>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
