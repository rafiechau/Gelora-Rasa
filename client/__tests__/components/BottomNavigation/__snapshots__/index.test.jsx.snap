// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomBar Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBottomNavigation-root css-16lloyr-MuiBottomNavigation-root"
    data-testid="bottom-navigation"
  />
</DocumentFragment>
`;
